app/src/main/java/org/mozilla/fenix/settings/biometric/BiometricPromptPreferenceFragment.kt

```diff
@@ -108,7 +107,10 @@ abstract class BiometricPromptPreferenceFragment : PreferenceFragmentCompat() {
         }
     }
 
-    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+    /**
+     * Use [BiometricPromptFeature] or [KeyguardManager] to handle pin request results.
+     */
+    fun pinRequestResult(requestCode: Int, resultCode: Int) {
```

Comment:
We could remove this function: it duplicates check for the `RESULT_OK` and there is not need for the request constant in the new API.

Reply:
This is currently being called in `AutofillSettingFragment.kt`.  Do you mean to move it?

Reply:
I am not sure, actually. What confuses me is, `AutofillSettingFragment` inherits `BiometricPromptPreferenceFragment`, and the child used to call`startActivityForResult` and the parent would receive the result. But the new API for `activityResult` doesn't use `requestCode` and is designed to handle the result in a lambda. We also handle `resultCode == RESULT_OK` inside the `registerForActivityResult` extension.
 
 Hm. If we want to keep the inheritance, how about we define `var startForResult: ActivityResultLauncher<Intent>` inside the parent, and children would just call `startForResult.launch(intent)`? Right now there is no other children, but it looks like the initial implementation of the `BiometricPromptPreferenceFragment` was aiming to handle `activityResult` in the parent, so we could keep the design.

Reply:
I think I see what you mean.  Make sense.  I'll update.  Thanks